packages/grafana-data/src/field/displayProcessor.ts

```diff
@@ -5,24 +5,35 @@ import _ from 'lodash';
 import { getColorFromHexRgbOrName } from '../utils/namedColorsPalette';
 
 // Types
-import { FieldConfig } from '../types/dataFrame';
+import { FieldConfig, FieldType } from '../types/dataFrame';
 import { GrafanaTheme, GrafanaThemeType } from '../types/theme';
 import { DisplayProcessor, DisplayValue, DecimalCount, DecimalInfo } from '../types/displayValue';
 import { getValueFormat } from '../valueFormats/valueFormats';
 import { getMappedValue } from '../utils/valueMappings';
 import { Threshold } from '../types/threshold';
-// import { GrafanaTheme, GrafanaThemeType, FieldConfig } from '../types/index';
+import { getTimeZoneDateFormatter } from '../datetime/moment_wrapper';
 
 interface DisplayProcessorOptions {
+  type?: FieldType;
```

Comment:
introducing field type to display processor options to be able to return date time formatter

Reply:
rather than adding `type` -- what do you think about replacing `config` with `field`?  then it would have access to other bits of the field if necessary too.

I'm not sure though.

Also torn on if `dateFormat` could/should be a setting on FieldConfig -- but make clearly a `dateDisplayFormat` or something.

Reply:
> rather than adding type -- what do you think about replacing config with field? then it would have access to other bits of the field if necessary too.

you mean referencing the entire field? Hmm, maybe a good idea, not sure. Can't see a use case for referencing field values and other bits here though. I like the current API when you pass config and get processor back. Especially that the processor is attached back to the field's `display` option.

Reply:
> Also torn on if dateFormat could/should be a setting on FieldConfig -- but make clearly a dateDisplayFormat or something.

Yeah, been on the fence about this a little bit. Should belong to `FieldConfig` probably. I'll change that